### PR TITLE
chore: add jest and github action

### PR DESCRIPTION
**Pull request by hazzadous**
We need to be able to ensure that the functionality is correct so we
make it possible to run `yarn jest` to validate.

### CODE REVIEW

File: index.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import { processEvent } from ".";
```
Comment (Burzo) [inline index.test.ts:1]: Please use prettier with the provided `.prettierrc` config file and format the file.

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
```
Comment (Burzo) [inline jest.config.js:1]: Same as `index.test.js` file above, please use `prettier` to format it.

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    preset: 'ts-jest',
+    moduleDirectories: ['node_modules'],
+    transform: {
+        '^.+\\.(ts|tsx)?$': 'ts-jest'
+    },
+    transformIgnorePatterns: [],
+    globals: {
+        'ts-jest': {
+            // Without isolatedModules, tests run realy slow, so we enable them
```
Comment (Burzo) [inline jest.config.js:10]: Nitpick: typo (`realy` -> `really`)

**Review by Burzo — CHANGES_REQUESTED** [review]

**Comment by Burzo** [comment]
Also, there is a deprecation warning when running tests:
```
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
```
which should be solvable by following the instructions in the warning.

**Review by Burzo — CHANGES_REQUESTED** [review]
There is also a deprecation warning reported by Jest which should be solvable by following the instructions in the warning itself:

```
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
```

**Comment by Burzo** [comment]
And also, hey @hazzadous! Hopefully, you didn't have to wait too long for us to take a look at this, and thanks for your help so far.

**Comment by hazzadous** [comment]
Hi @Burzo hope you get this working! I believe Xavier or Tiina at PostHog are the best to reference on any followups 🙏

**Pull request merged**